compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -780,23 +841,29 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         }
     }
 
+    /// Emits a [`StatementKind::StorageLive`] for the given var, and also
+    /// schedules a drop if requested (and possible).
     pub(crate) fn storage_live_binding(
         &mut self,
         block: BasicBlock,
         var: LocalVarId,
         span: Span,
         for_guard: ForGuard,
-        schedule_drop: bool,
+        schedule_drop: ScheduleDrops,
     ) -> Place<'tcx> {
         let local_id = self.var_local_id(var, for_guard);
         let source_info = self.source_info(span);
         self.cfg.push(block, Statement { source_info, kind: StatementKind::StorageLive(local_id) });
         // Although there is almost always scope for given variable in corner cases
         // like #92893 we might get variable with no scope.
-        if let Some(region_scope) = self.region_scope_tree.var_scope(var.0.local_id)
-            && schedule_drop
-        {
-            self.schedule_drop(span, region_scope, local_id, DropKind::Storage);
+        if let Some(region_scope) = self.region_scope_tree.var_scope(var.0.local_id) {
+            // Match exhaustively, so that it's easy to check how `ScheduleDrops` is used.
```

Comment:
I'm not sure I follow: do you find `matches!(schedule_drop, ScheduleDrops::Yes)` would make it harder to find?

Reply:
For very specific enums like this, that are basically defined by how they are consumed, I find it valuable to use an exhaustive match to express “the function of `ScheduleDrops::No` is to do nothing at this point”.

For example, if I use my IDE to search for references to `ScheduleDrops::No`, I get to see all the places that create that value, and all the places that consume that value, even if the way they consume that value is by (explicitly) doing nothing.

Reply:
Hm. That makes the code hard to scan for such a simple condition. I can merge it this way if you prefer, though I expect others to simplify it when they see it

Reply:
Hmm, while I do have a slight preference for my way, I think I'll change it to the simpler version just to avoid the ??? moment for other people. It's not a big deal either way.